Downgrade gson, Some files documented, Bugfix: Open file or directory does not work under mx-linux

woiddale opened the pull request:
Hallo Aclrian,
hier ist mein Pull Request.
Leider sind es drei Sachen auf einmal, aber vielleicht kannst du dir das mal anschauen.

1. In der module-info.java habe ich wie besprochen die gosn nochmal auf 2.10.1 zurückgesetzt. Außerdem habe ich ein paar "transitive" hinzugefügt, da ich hier Warnungen hatte.

2. Habe angefangen ein paar Dateien zu dokumentieren. Bin aber noch nicht fertig.

3. Habe einen BugFix in die Datei "src/main/java/net/aclrian/mpe/controller/MainController.java" eingebaut, da bei mir kein Programmordner, Messdienerordner oder Logdatei angezeigt wurde.
Hier stellt sich auch die Frage, was der Programmordner ist. Laut aktuellem Quellcode ist dies immer der Ordner "MessdienerplanErsteller" im Benutzerverzeichnis, egal wo sich das Programm tatsächlich befindet. Hier stellt sich mir die Frage, ob das so sein soll?

Viele Grüße
Walter

woiddale committed:
Downgrade to Gosn 2.10.1

woiddale committed:
Merge branch 'main' of https://github.com/Aclrian/MessdienerPlanErsteller

woiddale committed:
Downgrade gosn from 2.11.0 to 2.10.1

woiddale committed:
Angefangen zu dokumentieren.

woiddale committed:
Merge remote-tracking branch 'upstream/main'

woiddale committed:
Merge remote-tracking branch 'upstream/main' into doc_ASlider

woiddale committed:
Merge branch 'doc_ASlider'

woiddale committed:
Bugfix: Open file or directory does not work under mx-linux

woiddale committed:
Merge branch 'main' of https://github.com/Aclrian/MessdienerPlanErsteller

Aclrian reviewed (changes requested):
Danke für deinen Beitrag!

Ich habe mal etwas mehr angemerkt, damit der mir vor gestrebte einheitliche Style klar wird

woiddale commented on src/main/java/net/aclrian/mpe/utils/VersionIDHandler.java:
```diff
@@ -60,14 +79,25 @@ private static EnumVersionHandling getEnumVersionHandling() {
         return EnumVersionHandling.IS_THE_LATEST;
     }
 
+    /**
+     * Getter - Methode für internetId()
+     * @return internetId
+     */
```
Ok. Für mich stellt sich die Fragen, warum überhaupt die Methode getInternetId exisitert.

Aclrian replied:
So wie die Software historisch gewachsen ist (Es gab 2 andere UIs vorher) und ich mit ihr, kann es gut sein, dass solche Stellen vermehrt auftauchen, wenn alle Dateien einmal durchgeguckt werden. Falls du sowas siehst, kann das gerne auch angehen.

In diesem Fall hat meine IDE mich nicht darauf aufmerksam gemacht, da sie noch eine interne Verwendung hat.

Aclrian replied:
Umgesetzt mit: https://github.com/Aclrian/MessdienerPlanErsteller/commit/d5d9a480d379ec1437ae948cd8b70c3619dea4e8

woiddale commented on src/main/java/net/aclrian/mpe/controller/MainController.java:
```diff
@@ -226,22 +227,32 @@ public void info(ActionEvent actionEvent) {
+        String errorMessage = "Ihr Destkop unterstützt leider das Öffen nicht.\n" +
+                "Es sollte der Pfad:\n" + file + "\ngeöffnet werden.";
+
+        if(Desktop.isDesktopSupported()) {
+            new Thread(() -> {
+                try {
+                    Desktop.getDesktop().open(file);
+                }
+                catch (IOException e) {
+                        Dialogs.getDialogs().error(e, errorMessage);
+                }
+            }).start();
+        } else {
+            Dialogs.getDialogs().error(errorMessage);
+        }
```
Soweit habe ich die einzelnen Funktionen des Programms noch nicht getestet bzw. entdeckt.
Habe mir aber gerade mal die E-Mail und den Word - Export angesehen und beides funktioniert ebenfalls nicht.

Würde die Methode dann in eine eigene Klasse im Package net.aclrian.mpe.utils verschieben. Oder willst Du das machen?

Aclrian replied:
Ab April kann ich mehr machen, aber aktuell steht eine volle Prüfungsphase vor der Tür.

woiddale commented on src/main/java/net/aclrian/mpe/controller/MainController.java:
```diff
@@ -226,22 +227,32 @@ public void info(ActionEvent actionEvent) {
     }
 
+    private void desktopOpen(File file) {
+        String errorMessage = "Ihr Destkop unterstützt leider das Öffen nicht.\n" +
+                "Es sollte der Pfad:\n" + file + "\ngeöffnet werden.";
```
Die Methode wird zum Öffnen von Ordnern und Dateien verwendet. Deshalb habe ich das nicht einfach "Datei" genannt. Ebenso wollte ich den Benutzer darüber informieren, dass das Problem vermutlich mit seiner Desktopumgebung was zu tun hat.

woiddale commented on src/main/java/net/aclrian/mpe/utils/VersionIDHandler.java:
```diff
@@ -32,17 +45,23 @@ private static EnumVersionHandling rankingVersionID() {
             if (internetId.equals(MainApplication.VERSION_ID)) {
                 return EnumVersionHandling.IS_THE_LATEST;
             }
+            //Version aus dem Internet genauer prüfen.
```
Kann man sicherlich streichen. Aber der Aufruf "getEnumVersionHandling();" selber drückt nicht aus, was gemacht wird.
Du weißt natürlich sofort, was diese Funktion macht. Und das sie darunter steht. Aber jemand anders oder ich weiß das beim ersten Lesen nicht. Ebenso könnte irgendwann eine weitere Methode vor die Definition von getEnumVersionHandling rutschen und dann steht es nicht mehr direkt darunter.

Aber wie gesagt, wenn Du es gestrichen haben willst, kein Problem für mich.

Aclrian replied:
Dann sollte eher der Methodenname angepasst werden: z.B. `compareVersions()`

woiddale replied:
Habe ich gemacht.

Aclrian commented:
Bitte einmal den Branch updaten, sodass die Linter Fehler, die schon vorher drin waren behoben werden.

woiddale committed:
Merge branch 'main' of https://github.com/woiddale/MessdienerPlanErsteller

woiddale committed:
Merge branch 'main' of https://github.com/Aclrian/MessdienerPlanErsteller
Replaced tabs with 4 spaces

woiddale committed:
Kommentare überarbeitet

Verbesserte Kommentare von erstem Pull Request eingepflegt.

woiddale committed:
Nicht benutzte Elemente entfernt

Nicht benutzte Variablen entfernt.
Nicht benutzte import - Zeilen entfernt.

woiddale committed:
Desktop.getDesktop() - Aufrufe in Wrapper verschoben

Alle Aufruf open, mail und browse von Desktop.getDesktop() mit
DestkopWrapper realisiert.

woiddale committed:
Kommentare überarbeitet

Weitere Kommentare überarbeitet für PR.

woiddale commented:
Hi Aclrian,
habe soweit deine Vorschläge eingebaut und auch das mit den Desktop.getDesktop() aufrufen gemacht. Schau es dir bitte einfach mal an, wenn es deine Prüfungsvorbereitung zulässt oder dann halt nach den Prüfungen.

woiddale committed:
Private Konstruktor zur Klasse DesktopWrapper hinzugefügt.

woiddale commented:
MegaLinter Wartung beseitigt.

Aclrian committed:
Merge branch 'main' into main

Aclrian committed:
Merge branch 'main' into main

Aclrian reviewed (approved):
Vielen Dank!

woiddale commented:
Danke
